pygen/pygen_src/riscv_instr_sequence.py

```diff
@@ -54,9 +54,22 @@ def gen_instr(self, is_main_program, no_branch = 1):
             self.gen_stack_enter_instr()
             self.gen_stack_exit_instr()
 
-    # TODO
     def gen_stack_enter_instr(self):
-        pass
+        allow_branch = 0 if ((self.illegal_instr_pct > 0) or (self.hint_instr_pct > 0)) else 1
+        allow_branch &= not cfg.no_branch_jump
+        try:
+            with vsc.randomize_with(self.program_stack_len):
+                self.program_stack_len in vsc.rangelist(vsc.rng(cfg.min_stack_len_per_program,
+                                                                cfg.max_stack_len_per_program))
+                self.program_stack_len % (rcs.XLEN // 8) == 0
+        except Exception:
+            logging.critical("Cannot randomize program_stack_len")
+            sys.exit(1)
+        self.instr_stack_enter.push_start_label = self.label_name + "_stack_p"
+        self.instr_stack_enter.gen_push_stack_instr(self.program_stack_len,
+                                                    allow_branch = allow_branch)
+        self.instr_stream.instr_list.extend((self.instr_stack_enter.instr_list,
+            self.instr_stream.instr_list))
```

Comment:
These two lines won't work as `extend` intends, it will append two separate "lists" to the `self.instr_stream.instr_list`. Just to confirm, is that intentional?  If not, perhaps we should replace it with:
`self.instr_stream.instr_list.extend(self.instr_stack_enter.instr_list)
self.instr_stream.instr_list.extend(self.instr_stream.instr_list)`

Reply:
Hi @hodjat91 
This is not intensional, In fact, the second list is inappropriate to use here.
Just realize after looking at your comment,
Thanks for this
I will fix it right away,